src/averroes/Main.java

```diff
@@ -63,29 +68,60 @@ public static void main(String[] args) {
 			System.out.println("Organizing the JAR files ...");
 			JarOrganizer jarOrganizer = new JarOrganizer();
 			jarOrganizer.organizeInputJarFiles();
-
+			
 			// Print some statistics
-			System.out.println("# application classes: " + jarOrganizer.applicationClassNames().size());
+			if (AverroesOptions.isAndroid()) {	
+				System.out.println("# referenced application classes: " + SetupAndroid.v().getReferencedApplicationClassCount());
+				System.out.println("# referenced application methods: " + SetupAndroid.v().getReferencedApplicationMethodCount());
+			}
+			else {	
+				System.out.println("# application classes: " + jarOrganizer.applicationClassNames().size());
+			}
 			System.out.println("# library classes: " + jarOrganizer.libraryClassNames().size());
 
 			// Add the organized archives for the application and its
 			// dependencies.
 			TimeUtils.reset();
 			JarFactoryClassProvider provider = new JarFactoryClassProvider();
 			provider.prepareJarFactoryClasspath();
-
-			// Set some soot parameters
+			
+			//Set some soot parameters for android
+            if(AverroesOptions.isAndroid())
+            {
+            	List<ClassProvider> classProviders = new LinkedList<>();
+    			classProviders.add((ClassProvider) provider);	
+    			classProviders.add(new DexClassProvider());
+    			SourceLocator.v().setClassProviders(classProviders);
+    			SootSceneUtil.addCommonDynamicClasses(provider);
+    			
+    			SetupAndroid setupAndroid = null;
+    			SootMethod dummyMain = null;
+    			if (AverroesOptions.isAndroid()) {	
+    				setupAndroid = SetupAndroid.v();
+    				dummyMain = setupAndroid.getDummyMainMethod(); 
+     			}
+    			else {
+    				Options.v().classes().addAll(provider.getApplicationClassNames());
+    			}	
+            }
+            // Set some soot parameters if not android 
+            else if (!AverroesOptions.isAndroid()){
 			SourceLocator.v().setClassProviders(Collections.singletonList((ClassProvider) provider));
 			SootSceneUtil.addCommonDynamicClasses(provider);
 			Options.v().classes().addAll(provider.getApplicationClassNames());
 			Options.v().set_main_class(AverroesOptions.getMainClass());
+            }
 			Options.v().set_validate(true);
 
 			// Load the necessary classes
 			System.out.println("");
 			System.out.println("Loading classes ...");
 			Scene.v().loadNecessaryClasses();
-			Scene.v().setMainClassFromOptions();
+			
+			//only in case of java files
```

Comment:
either bring the if-body to the same line or surround it with curly braces (even if it's a single statement)